aws/hll/aws-s3-transfer-manager/src/io/part_reader.rs

```diff
@@ -0,0 +1,336 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+use std::cmp;
+use std::ops::DerefMut;
+use std::sync::Mutex;
+
+use bytes::{Buf, Bytes, BytesMut};
+
+use crate::io::error::Error;
+use crate::io::path_body::PathBody;
+use crate::io::stream::RawInputStream;
+use crate::io::InputStream;
+use crate::MEBIBYTE;
+
+/// Builder for creating a `ReadPart` implementation.
+#[derive(Debug)]
+pub(crate) struct PartReaderBuilder {
```

Comment:
I have a preference for giving stuff like this a generic `Builder` name and then referring to it with the module.
```suggestion
pub(crate) struct Builder {
```
We'd then say `part_reader::Builder` to refer to it.